exercises/change/example.go

```diff
@@ -0,0 +1,191 @@
+// Package change demonstrates making change with fewest number of coins.
+package change
+
+import (
+	"errors"
+	"sort"
+)
+
+const testVersion = 1
+
+var (
+	ErrNoSolutionFound = errors.New("No solution")
+	ErrNegativeTarget  = errors.New("Negative Target")
+)
+
+// Change returns a list of coins in increasing value order to make
+// change for target amount from the coins given; an infinite supply of coins is assumed.
+func Change(coins []int, target int) (result []int, err error) {
+	var solution []int
+	if target == 0 {
+		// No change needed for 0.
+		return []int{}, nil
+	}
+	if target < 0 {
+		// No change possible for negative.
+		return nil, ErrNegativeTarget
+	}
+	findSolution(coins, target, &solution)
+	if len(solution) == 0 {
+		// No solution possible with coins given.
+		return nil, ErrNoSolutionFound
+	}
+	return solution, nil
+}
+
+// findSolution attempts to find a change solution for target amount
+// from the coins given.
+func findSolution(coins []int, target int, s *[]int) {
+	// Find the coins to consider, less than the target change amount.
+	last := len(coins) - 1
+	for ; last >= 0; last-- {
+		if coins[last] == target {
+			// Perfect single coin solution, so done!
+			*s = []int{target}
+			return
+		}
+		if coins[last] > target {
+			// Cannot use this too large coin.
+			continue
+		}
+		// Now, 0 .. last are the coins to consider.
+		break
+	}
+	if last < 0 {
+		// All coins larger than target.
+		return
+	}
+	findBestMixedSolution(coins[0:last+1], target, s)
+}
+
+// findBestMixedSolution attempts to find a best mixed coin amount solution
+// making correct change for the amount target.
+func findBestMixedSolution(coins []int, target int, s *[]int) {
+	// Consider all combinations of the coins.
```

Comment:
After experimenting: Looks like it works to me, after changing `k >= 1` to `k >= 0`. At this point, `findBestMixedSolution` isn't even necessary. You can let `findSolution` call `partialSolution`. Suggest you give that a try.